node/authorship/src/authorship.rs

```diff
@@ -520,25 +549,87 @@ where
 
         self.transaction_pool.remove_invalid(&unqueue_invalid);
 
-        // Pushing a custom extrinsic that must run at the end of a block
-        let custom_extrinsic = block_builder.create_terminal_extrinsic(self.max_gas)?;
+        // Attempt to apply pseudo-inherent on top of the current overlay in a separate thread.
+        // In case the timeout is hit, we will proceed without it.
+        let client = self.client.clone();
+        let backend = self.backend.clone();
+        let pseudo_inherent = block_builder.create_terminal_extrinsic(self.max_gas)?;
```

Comment:
shouldn't we create it already in update_block?

Reply:
Eventually, got rid of this runtime api call at all by introducing an additional method `push_final` in the `BlockBuilder` which does both things: fetches the pseudo-inherent extrinsic (a.k.a. `Gear::run()`) and tris to apply it inside a single transactional call.